src/fmu/dataio/dataio.py

```diff
@@ -865,6 +865,11 @@ def _export_without_metadata(self, obj: types.Inferrable) -> str:
     def _export_with_product(self, obj: types.Inferrable, product: Product) -> str:
         """Export the object with product information in the metadata."""
 
+        if not isinstance(self.config, GlobalConfiguration):
+            raise ValidationError(
+                "When exporting standard_results it is required to have a valid config."
```

Comment:
Kept this rather simple as there will be several warnings emitted during ExportData initialization highlighting the actual validation errors and/or directing users towards the `Getting started guide`